File: Stack_Problems/RemoveStarsFromString.java

// Problem link - https://leetcode.com/problems/removing-stars-from-a-string/description/

package Stack_Problems;

import java.util.Stack;

public class RemoveStarsFromString {
    public static String removeStars(String s) {
        Stack<Character> stack = new Stack<>();

        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) != '*'){
                stack.push(s.charAt(i));
            }else{
                if(!stack.isEmpty())
                    stack.pop();
            }
        }

        StringBuilder res = new StringBuilder();
        while (!stack.isEmpty()) {
            res.insert(0, stack.pop());
        }
        return res.toString();
    }

    public static void main(String[] args) {
        String str = "leet**cod*e";
        System.out.println(removeStars(str));
    }
}

File: 07.Stack_Problems/RemoveStarsFromString.js

// Problem link - https://leetcode.com/problems/removing-stars-from-a-string/description/

let removeStars = function(s) {
    let stack = [];
    for(let i=0; i<s.length; i++){
        if(s[i] !== '*'){
            stack.push(s[i]);
        }else{
            stack.pop();
        }
    }
    return stack.join('');
};

let str = "leet**cod*e";
let ans = removeStars(str);
console.log(ans);